src/tests/test_console.py

```diff
@@ -94,3 +99,84 @@ def stderr_callback(string):
 
     print("bar")
     assert my_stdout == "foobar\n"
+
+    shell.push("print('foobar')")
+    assert my_stdout == "foobar\nfoobar\n"
+
+    shell.push("import sys")
+    shell.push("print('foobar', file=sys.stderr)")
+    assert my_stderr == "foobar\n"
+
+    shell.push("1+1")
+    assert my_stdout == "foobar\nfoobar\n2\n"
+
+
+def test_repr(safe_sys_redirections):
+    assert len(console.repr_abreviate("x" * 10 ** 5, 100)) <= 110
+
+    sys.stdout = io.StringIO()
+    console.displayhook([0] * 100, lambda v: console.repr_abreviate(v, 100))
+    assert len(sys.stdout.getvalue()) <= 110
+
+
```

Comment:
It would be good to add tests not using pyodide for `repr_abreviate` (or the other name we give it). In particular, with a small `limit` (say around 10) checking the output is correct for odd and even combination of input lenght and limit values.